lms/djangoapps/letstudy_api/views.py

```diff
@@ -0,0 +1,165 @@
+            username = self._check_available_required_params(request.data.get('username'), "username")
+            uid = self._check_available_required_params(request.data.get('uid'), "uid")
+            if check_account_exists(username=username, email=email):
+                return Response(data={"error_message": "User already exists"}, status=status.HTTP_409_CONFLICT)
+            if UserSocialAuth.objects.filter(uid=uid).exists():
+                return Response(
+                        data={"error_message": "Parameter 'uid' isn't unique."},
+                        status=status.HTTP_409_CONFLICT
+                )
+            data['name'] = "{} {}".format(first_name, last_name).strip() if first_name or last_name else username
+            data['first_name'] = first_name
+            data['last_name'] = last_name
+            data['password'] = uuid4().hex
+            for param in ('is_active', 'allow_certificate'):
+                data[param] = data.get(param, True)
+            user = User.objects.create_user(username=username, email=email, password=data['password'])
+            UserProfile.objects.create(user=user)
+            idp_name = OAuth2ProviderConfig.objects.first().backend_name
+            UserSocialAuth.objects.create(user=user, provider=idp_name, uid=uid)
+        except (ValueError, ValidationError) as e:
+            return Response(
+                data={"error_message": e.message},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        return Response(data={'user_id': user.id, 'username': username}, status=status.HTTP_200_OK)
+
+
+class UserEnrollView(APIView, ApiKeyPermissionMixIn):
+    """
+    API endpoint for enrolling user on the course.
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    def post(self, request):
+        """
+        Enroll user on the course.
+
+        :param course: Course id. string, in the format {key type}:{org}+{course}+{run}.
+                                                        For example, course-v1:edX+DemoX+Demo_2014.
+        :param user_email: User email string
+        :param request: request for Api endpoint
+        :param action: string 'enroll'
+        :param mode: string indicating what kind of enrollment this is: audit or verified.
+        """
+        data = request.data
+        user_email = data.get('user_email')
+        mode = data.get('mode')
+        course_id = data.get('course')
+        action = data.get('action')
+        if action != 'enroll':
+            return Response(
+                data={"error_message": "Action must be enroll"},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        try:
+            user = get_student_from_identifier(user_email)
+
+            if get_enrollment(user, course_id):
+                return Response(
```

Comment:
I think it should be better to return Response in one point, and set different data and status in various cases

Reply:
fixed